scipy/ndimage/tests/test_filters.py

```diff
@@ -86,6 +87,173 @@ def test_valid_origins():
         assert_raises(ValueError, filter, data, 3, origin=2)
 
 
+def test_multiple_modes():
+    # Test that the filters with multiple mode cababilities for different
+    # dimensions give the same result as applying a single mode.
+    arr = np.zeros((10, 10), float)
+    arr[5, 5] = 1
+    arr[5, 0] = 1
+    arr[0, 0] = 1
+
+    mode1 = 'reflect'
+    mode2 = ['reflect', 'reflect']
+
+    assert_equal(sndi.gaussian_filter(arr, 1, mode=mode1),
+                 sndi.gaussian_filter(arr, 1, mode=mode2))
+    assert_equal(sndi.prewitt(arr, mode=mode1),
+                 sndi.prewitt(arr, mode=mode2))
+    assert_equal(sndi.sobel(arr, mode=mode1),
+                 sndi.sobel(arr, mode=mode2))
+    assert_equal(sndi.laplace(arr, mode=mode1),
+                 sndi.laplace(arr, mode=mode2))
+    assert_equal(sndi.gaussian_laplace(arr, 1, mode=mode1),
+                 sndi.gaussian_laplace(arr, 1, mode=mode2))
+    assert_equal(sndi.maximum_filter(arr, size=5, mode=mode1),
+                 sndi.maximum_filter(arr, size=5, mode=mode2))
+    assert_equal(sndi.minimum_filter(arr, size=5, mode=mode1),
+                 sndi.minimum_filter(arr, size=5, mode=mode2))
+    assert_equal(sndi.gaussian_gradient_magnitude(arr, 1, mode=mode1),
+                 sndi.gaussian_gradient_magnitude(arr, 1, mode=mode2))
+    assert_equal(sndi.uniform_filter(arr, 5, mode=mode1),
+                 sndi.uniform_filter(arr, 5, mode=mode2))
+
+
+def test_multiple_modes_sequentially():
+    # Test that the filters with multiple mode cababilities for different
+    # dimensions give the same result as applying the filters with
+    # different modes sequentially
+    arr = np.zeros((10, 10), float)
+    arr[5, 5] = 1
+    arr[5, 0] = 1
+    arr[0, 0] = 1
+
+    modes = ['reflect', 'wrap']
+
+    expected = sndi.gaussian_filter1d(arr, 1, axis=0, mode=modes[0])
+    expected = sndi.gaussian_filter1d(expected, 1, axis=1, mode=modes[1])
+    assert_equal(expected,
+                 sndi.gaussian_filter(arr, 1, mode=modes))
+
+    expected = sndi.uniform_filter1d(arr, 5, axis=0, mode=modes[0])
+    expected = sndi.uniform_filter1d(expected, 5, axis=1, mode=modes[1])
+    assert_equal(expected,
+                 sndi.uniform_filter(arr, 5, mode=modes))
+
+    expected = sndi.maximum_filter1d(arr, size=5, axis=0, mode=modes[0])
+    expected = sndi.maximum_filter1d(expected, size=5, axis=1, mode=modes[1])
+    assert_equal(expected,
+                 sndi.maximum_filter(arr, size=5, mode=modes))
+
+    expected = sndi.minimum_filter1d(arr, size=5, axis=0, mode=modes[0])
+    expected = sndi.minimum_filter1d(expected, size=5, axis=1, mode=modes[1])
+    assert_equal(expected,
+                 sndi.minimum_filter(arr, size=5, mode=modes))
+
+
+def test_multiple_modes_prewitt():
+    # Test prewitt filter for multiple extrapolation modes
+    arr = np.zeros((3, 3), float)
+    arr[1, 1] = 1
+    arr[1, 0] = 1
+    arr[0, 0] = 1
```

Comment:
This would be more readable if constructed like the others explicitly as `np.array([...], [...], [...])`